renderdoc/core/resource_manager.h

```diff
@@ -675,6 +706,19 @@ void ResourceManager<Configuration>::MarkResourceFrameReferenced(ResourceId id,
   if(id == ResourceId())
     return;
 
+  if(IsDirtyFrameRef(refType))
+  {
+    // If the resource was postponed during Active Capture, we need to prepare it
+    // right away, since next Read might be invalid.
+    if(IsResourcePostponed(id))
+    {
+      RDCDEBUG("Preparing resource %llu after it has been postponed.", id);
+      Prepare_ResourceInitialStateIfNeeded(id);
+    }
+
+    UpdateLastWriteTime(id);
```

Comment:
@baldurk, good points, thanks. Yeah, I was actually thinking to expose the capture state to the base resource manager (with a default getter, when configurations can override providing their own state).

And yes, we'll need to clean up the last write time in the `ReleaseCurrentResource`.

Reply:
I think at the moment while some of the API implementations of resource manager do have a local `CaptureState` they only use it to differentiate capture from replay, none of them update based on background/active capturing.

The best way to do it is probably to initialise the ResourceManager with a `CaptureState &state` since we can pass through the owning API's variable when creating the resource manager. Then since it's a reference it will always be up to date either background or active capturing.